### PR TITLE
Cvar migration: cgame numeric vars

### PR DESCRIPTION
**Pull request by slipher**
Running my cvar migration script some more. This time focused on non-string cgame cvars.

### CODE REVIEW

**Comment by slipher** [comment]
Note to self: check if I need to mark a bunch of things "archive"...

**Comment by slipher** [comment]
Double-checked that the archiving behavior is no different with the new style.

This is ready (other than a few cvars having "I don't know" as the description, but I don't plan to fix that).

File: src/cgame/cg_main.cpp [hunk 1]
```diff
@@ -39,133 +39,129 @@ upgradeInfo_t   cg_upgrades[ 32 ];
...
+Cvar::Modified<Cvar::Cvar<std::string>> cg_rangeMarkerBuildableTypes("cg_rangeMarkerBuildableTypes", "list of buildables or buildable types to show range marker for", Cvar::NONE, "support");
+Cvar::Cvar<bool> cg_rangeMarkerWhenSpectating("cg_rangeMarkerWhenSpectating", "show buildable rangers while spectating", Cvar::NONE, false);
+int cg_buildableRangeMarkerMask;
+Cvar::Range<Cvar::Cvar<float>> cg_binaryShaderScreenScale("cg_binaryShaderScreenScale", "I don't know", Cvar::NONE, 1.0, 0, 1);
+
+Cvar::Cvar<float> cg_painBlendUpRate("cg_painBlendUpRate", "I don't know", Cvar::NONE, 10.0);
+Cvar::Cvar<float> cg_painBlendDownRate("cg_painBlendDownRate", "I don't know", Cvar::NONE, 0.5);
+Cvar::Cvar<float> cg_painBlendMax("cg_painBlendMax", "I don't know", Cvar::NONE, 0.7);
+Cvar::Cvar<float> cg_painBlendScale("cg_painBlendScale", "I don't know", Cvar::NONE, 7.0);
+Cvar::Cvar<float> cg_painBlendZoom("cg_painBlendZoom", "I don't know", Cvar::NONE, 0.65);
```
Comment (necessarily-equal) [inline src/cgame/cg_main.cpp:127]:
```suggestion
Cvar::Cvar<float> cg_painBlendUpRate("cg_painBlendUpRate", "how fast the pain indicator will appear", Cvar::NONE, 10.0);
Cvar::Cvar<float> cg_painBlendDownRate("cg_painBlendDownRate", "how fast the pain indicator will disappear", Cvar::NONE, 0.5);
Cvar::Cvar<float> cg_painBlendMax("cg_painBlendMax", "upper bound on how opaque the pain indicator will be", Cvar::NONE, 0.7);
Cvar::Cvar<float> cg_painBlendScale("cg_painBlendScale", "how amplified the damage will be for the blood indicator (1->damage is barely visible, 20->damage reaches cg_painBlendMax almost instantly", Cvar::NONE, 7.0);
Cvar::Cvar<float> cg_painBlendZoom("cg_painBlendZoom", "size scale factor for the the pain indicator", Cvar::NONE, 0.65);
```

File: src/cgame/cg_main.cpp [hunk 2]
```diff
@@ -480,17 +356,11 @@ void CG_UpdateBuildableRangeMarkerMask()
...
-		int brmMask = cg_rangeMarkerWhenSpectating.integer ? ( 1 << BA_NONE ) : 0;
+		int brmMask = 0;
 
-		if ( !cg_rangeMarkerBuildableTypes.string[ 0 ] )
-		{
-			goto empty;
-		}
-
-		for (Parse_WordListSplitter marker(cg_rangeMarkerBuildableTypes.string); *marker; ++marker)
+		for (Parse_WordListSplitter marker(*std::move(cg_rangeMarkerBuildableTypes)); *marker; ++marker)
```
Comment (necessarily-equal) [inline src/cgame/cg_main.cpp:363]: I'm not sure this is correct: how can you move the cvar? How is this different from 
```suggestion
		for (Parse_WordListSplitter marker(std::move(cg_rangeMarkerBuildableTypes.Get())); *marker; ++marker)
```

File: src/cgame/cg_main.cpp [hunk 3]
```diff
@@ -39,133 +39,129 @@ upgradeInfo_t   cg_upgrades[ 32 ];
...
-vmCvar_t        cg_animBlend;
+Cvar::Cvar<bool> cg_animSpeed("cg_animspeed", "run animations? (for debugging)", Cvar::CHEAT, true);
+Cvar::Cvar<float> cg_animBlend("cg_animblend", "I don't know", Cvar::NONE, 5.0);
 
-vmCvar_t        cg_motionblur;
-vmCvar_t        cg_motionblurMinSpeed;
-vmCvar_t        cg_spawnEffects;
+Cvar::Cvar<float> cg_motionblur("cg_motionblur", "strength of motion blur", Cvar::NONE, 0.05);
+Cvar::Cvar<float> cg_motionblurMinSpeed("cg_motionblurMinSpeed", "speed triggering motion blur", Cvar::NONE, 600);
+Cvar::Cvar<bool> cg_spawnEffects("cg_spawnEffects", "world is gray when dead or spawning", Cvar::NONE, true);
```
Comment (necessarily-equal) [inline src/cgame/cg_main.cpp:152]:
```suggestion
Cvar::Cvar<float> cg_motionblurMinSpeed("cg_motionblurMinSpeed", "minimum speed to trigger motion blur", Cvar::NONE, 600);
Cvar::Cvar<bool> cg_spawnEffects("cg_spawnEffects", "desaturate world view when dead or spawning", Cvar::NONE, true);
```

File: src/cgame/cg_main.cpp [hunk 4]
```diff
@@ -39,133 +39,129 @@ upgradeInfo_t   cg_upgrades[ 32 ];
...
+Cvar::Cvar<float> cg_gun_z("cg_gunZ", "model debugging: gun z offset", Cvar::CHEAT, 0);
+Cvar::Cvar<bool> cg_mirrorgun("cg_mirrorgun", "use left-handed gun", Cvar::NONE, false);
+Cvar::Range<Cvar::Cvar<float>> cg_tracerChance("cg_tracerchance", "probability to draw line on bullet trajectory", Cvar::CHEAT, 1.0f, 0.0f, 1.0f);
+Cvar::Cvar<float> cg_tracerWidth("cg_tracerwidth", "width of line on bullet trajectory", Cvar::CHEAT, 3);
+Cvar::Cvar<float> cg_tracerLength("cg_tracerlength", "length of line drawn on bullet trajectory", Cvar::CHEAT, 200);
+Cvar::Cvar<bool> cg_thirdPerson("cg_thirdPerson", "show own player from 3rd-person perspective", Cvar::CHEAT, false);
+Cvar::Cvar<float> cg_thirdPersonAngle("cg_thirdPersonAngle", "yaw angle for 3rd-person view", Cvar::CHEAT, 0);
+Cvar::Range<Cvar::Cvar<int>> cg_thirdPersonShoulderViewMode("cg_thirdPersonShoulderViewMode", "alternative chase cam position", Cvar::NONE, 1, 1, 2);
+Cvar::Cvar<bool> cg_staticDeathCam("cg_staticDeathCam", "don't follow attacker movements after death", Cvar::NONE, false);
+Cvar::Cvar<bool> cg_thirdPersonPitchFollow("cg_thirdPersonPitchFollow", "I don't know", Cvar::NONE, false);
```
Comment (necessarily-equal) [inline src/cgame/cg_main.cpp:85]:
```suggestion
Cvar::Cvar<bool> cg_thirdPersonPitchFollow("cg_thirdPersonPitchFollow", "do follow the view pitch of the player you follow (disabled by default for comfort)", Cvar::NONE, false);
```

**Review by necessarily-equal — COMMENTED** [review]
I've read it fast enough, but on principle I think we can just merge and iron out the bugs afterwards. I've fond one odd thing and added some text otherwise.

File: src/cgame/cg_main.cpp (hunk 2 above)
Comment (slipher) [reply src/cgame/cg_main.cpp:363]: Good catch, that was supposed to say `structureList`.

**Pull request merged**